src/sidebar/frame-sync.js

```diff
@@ -126,12 +128,24 @@ function FrameSync($rootScope, $window, Discovery, annotationUI, bridge) {
 
     bridge.on('destroyFrame', destroyFrame.bind(this));
 
+    // Map of annotation tag to anchoring status
+    // ('anchored'|'orphan'|'timeout').
+    //
+    // Updates are coalesced to reduce the overhead from processing
+    // triggered by each `UPDATE_ANCHOR_STATUS` action that is dispatched.
+    var anchoringStatusUpdates = {};
+    var scheduleAnchoringStatusUpdate = debounce(() => {
+      annotationUI.updateAnchorStatus(anchoringStatusUpdates);
+      $rootScope.$broadcast(events.ANNOTATIONS_SYNCED, Object.keys(anchoringStatusUpdates));
+      anchoringStatusUpdates = {};
+    }, 10);
```

Comment:
Ok, so the anonymous function passed to `debounce()` will be called at most once every 10ms, no matter how many times we try to call it. And when it does finally get called, it uses `anchoringStatusUpdates` to update `annotationUI` and broadcast `ANNOTATIONS_SYNCED`, and then it resets `anchoringStatusUpdates` to empty.

Reply:
Yes, exactly.